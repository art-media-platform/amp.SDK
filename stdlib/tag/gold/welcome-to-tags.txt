
Hello World!
Hello, World!
hello  world
world  hello
World, hello.

Hello and welcome to tags!
hello, and welcome to tags!
hello, and Welcome to tags!!
hello, and..welcome to... tags.
hello+and+welcome+to+tags


UTF8 whitespace, commas, +, and . and : denote the "with" operator.   This operator adds the hash of each operand and is also therefore commutative.

Text having "homophone case" or "homophonic case" means its component words are considered interchangeable with words that sound the same but have different spelling.

The words Center, center, centre, and Centre are homophonic and should have the same hash.
The words center, Center, Centre, and centre are homophonic and should have the same hash.
The words Centre, centre, center, and Center are homophonic and should have the same hash.

This has very helpful implications for those with sight or hearing impairments, or in search result ranking since the query word order is usually less important the presence of particular words, especially once there are dozen words or more.

However: dashes and slashes are order dependent!?  Yes, order-dependent!
However, dashes and slashes are order dependent?!  yes, order-dependent
However, Slashes and dashes are order dependent?  Yes, order-dependent
However: Order-dependent?  Yes, slashes, and dashes are order dependent.
however: dashes and slashes are order-dependent! yES! Order dependent!

However: dependent-order?  Yes, slashes and dashes are order dependent.
However: Slashes and dashes are order dependent. Yes, dependent-order?
-order dependent, yes. Slashes and dashes are order dependent.  However.

Acronyms like NBA, NFL, USA, CIA, PRC, and EU are capitalized since they are spoken letter by letter.

Words that are not upper case, are considered spoken or otherwise phonetic.

This also means we can communicate proper nouns and titles easily, such as A Game of Thrones
This also means we can communicate proper nouns and titles easily, such as Game of Thrones, A.
This also means we can communicate proper nouns and titles easily, such as a Game Of Thrones.

Notice how acronyms are spoken differently and have their own unique meanings, so that means ALL-CAPS should be used intentionally.

ADA is an acronym for the Americans with Disabilities Act, Ms. Ada is a computer programmer.
ADA is an acronym for the Americans with Disabilities Act. ms ada is a computer programmer!
ADA is an acronym for the Americans with Disabilities Act. Ada is a computer programmer, Ms!
ADA is an acronym for the Americans With Disabilities Act, A computer programmer, Ms. ada is.

Dig into the amp SDK today!
Dig into the amp.SDK today!
Dig into the Amp SDK today!


Yoda said this is not a drill.
Yoda said, a drill this is not.
Yoda said, a drill this is not!
a drill this is not, Yoda said!
said a drill, this is not Yoda.
said a drill, is this not Yoda?
said a drill: Is this not Yoda?
Said a drill,: Is this not Yoda??


Dots are the canonical way, ya know.  after a dozen words, 24 bytes is 2^192!
Ya know, dots are the canonical way.  after a dozen words, 24 bytes is 2^192!


The 24-byte tag.ID you see to the left is 2^192 wide (in base 32).  For reference, a Bitcoin address is 20 bytes.

Base16 of the ID above (8kj00c8gw8406tugx9bx417whqj3y0qkg) is 160 bits wide.  The leading 32 bits are part of a "minting" when a tag is made decidedly private.

The number of possible hashes for a string of 10 words randomly chosen from the worlds dictionaries well over a million words. Even with the English dictionary's 200,000 words, there are 10^50 (2^160) outputs.


TODO:   go get github.com/alecthomas/participle/v2 -- for tags.Parse() implementation?
