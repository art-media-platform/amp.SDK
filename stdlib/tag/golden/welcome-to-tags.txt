
Hello World!
Hello, World!
hello  world
world  hello
World, hello.

Hello and welcome to tags!
hello, and welcome to tags!
hello, and Welcome to tags!!
hello, and..welcome to... tags.
hello+and+welcome+to+tags


UTF8 whitespace, commas, +, and . denote the "with" operator.   This operator adds the hash of each operand and is also therefore commutative.

Text having "homophone case" or "homophonic case" means its component words are considered interchangeable with words that sound the same but have different spelling.

The words Center, center, centre, and Centre are homophonic.
The words center, Center, Centre, and centre are homophonic.
The words Centre, centre, center, and Center are homophonic.

This has very helpful implications for those with sight or hearing impairments, or in search result ranking since the query word order is usually less important the presence of particular words, especially once there are dozen words or more.

However, colons, dashes and slashes are order dependent!?  Yes, order-dependent!
However, colons, dashes and slashes are order dependent?!  yes, order-dependent
However, Slashes, colons and dashes are order dependent?  Yes, order-dependent
However, Order-dependent?  Yes, colons, slashes, and dashes are order dependent.
however, dashes, colons and slashes are order-dependent! yES! Order dependent!

Acronyms like NBA, NFL, USA, CIA, PRC, and EU are capitalized since they are spoken letter by letter.

Words that are not upper case, are considered spoken / phonetic.

This also means we can communicate proper nouns and titles easily, such as A Game Of Thrones
This also means we can communicate proper nouns and titles easily, such as Game of Thrones, A.
This also means we can communicate proper nouns and titles easily, such as a game of thrones.

Notice how acronyms are spoken differently and have their own unique meanings, so that means ALL-CAPS should be used intentionally.

ADA is an acronym for the Americans with Disabilities Act, Ms. Ada is a computer programmer.
ADA is an acronym for the Americans with Disabilities Act. ms ada is a computer programmer!
ADA is an acronym for the Americans with Disabilities Act. Ada is a computer programmer, Ms!
ADA is an acronym for the Americans With Disabilities Act, A computer programmer, Ms. ada is.

Get into the amp SDK today!
Get into the amp.SDK today!
Get into the Amp SDK today!


Yoda said this is not a drill.
Yoda said, a drill this is not.
Yoda said, a drill this is not!
a drill this is not, Yoda said!
said a drill, this is not Yoda.
said a drill, is this not Yoda?

12 words randomly chosen from, but not limited to, the English dictionary exceeds 10^56 possible combinations.  For reference, that's over 10 million times the number of possible Bitcoin public addresses (2^160).

The above hash ID G5NRUJZEX37FZE7J38C0PRSQNCGF8DCXB demonstrates a standardized way to precisely witness a set of unordered literals. In creating a subsequent hash ID, we are cryptographically chaining tag sets together.

And so by citing both G5NRUJZEX37FZE7J38C0PRSQNCGF8DCXB and BSJ6041BGWJM08K4S0M05995JZSQRHPV2 as literals, we can conveniently validate tag relationships and groups.

Hashing 9X75KTQ662RMTKRTNF0FY2PSWUC4N21G3 with H3 geo-tiles 8D489E34600637F and 872AA845AFFFFFF demonstrates how locations can be cryptographically bound together in this system.  This offers easy "DIY" layering of security and permissions.